plugins/scaffolder-backend/src/scaffolder/tasks/types.ts

```diff
@@ -211,6 +211,16 @@ export interface TaskStore {
   ): Promise<{ events: SerializedTaskEvent[] }>;
 
   shutdownTask?(options: TaskStoreShutDownTaskOptions): Promise<void>;
+
+  getWorkspace?(options: { taskId: string }): Promise<Buffer | undefined>;
```

Comment:
Might be good to make this interface take a target path instead of returning a buffer. This sets us up for being able to unpack them in a streaming way in the future without interface changes, alleviating the need for storing all of the data in memory in-between.

Reply:
@freben Just to make sure that I got you correctly.
As this is an API which fetches the workspace from the database.

The suggestion is to modify it into this?:

```
  async getWorkspace?(options: {
    taskId: string;
  }): Promise<string> {
    const [result] = await this.db<RawDbTaskRow>('tasks')
      .where({ id: options.taskId })
      .select('workspace');

   const workspacePath = path.join(this.options.workingDirectory, options.taskId);
   await restoreWorkspace(workspacePath, result.workspace);

    return workspacePath;
  }
```

Reply:
When we will align on this one, I'll create a new MR. 
Merging master into the branch messed this MR up.

Reply:
Or like

```js
  async getWorkspace?(options: {
    taskId: string;
    targetPath: string;
  }): Promise<void>;
```

Letting the caller decide where to place the output (expected to be an existing, preferably empty directory).

Then if it's _internally implemented_ to load something into memory and then unpacking it, it's at least hidden from our view.

Does that make sense? This is just an idea, you guys get to decide.